add VIA format

gaieges opened the pull request:
Added a new VIA data defintion model for use in various places to validate that the VIA results are accurate.  I had this lying around for something I was doing and figured I'd at least get it saved somewhere.  It'll likely need some evolution in the near future.

I had to move around some of the stuff in `__init__` to the manifest.py file, but kept all the imports for backward compatibility.  None of that code changes though.

e271828- commented:
Please change this to "via"

todicus commented on basemodels/vgg.py:
```diff
@@ -0,0 +1,54 @@
+        },
+        "class_attributes": {
+            "0": {
+                "class_attributes": {
```
I'm concerned this nested `class_attributes` will lead to confusion in the future. I know I brought this up the other day, but I just dug a little more and I don't think this is part of the VGG format. VGG doesn't really concern itself with global image classification, just regions within an image.

In [manifestgenerator.py](https://github.com/IntuitionMachines/loader/blob/14ab5ba10ce79cf8a66573738289530242b07bf3/manifestgenerator.py#L469) we note that "class_attributes contains the labels that are associated with each whole image". The closest VGG field is `file_attributes`.


Reference VGG output JSON:
```
{
	"wikimedia_death_of_socrates.jpg-1": {
		"file_attributes": {
			"caption": "The Death of Socrates by David",
			"image_url": "https://en.wikipedia.org/wiki/The_Death_of_Socrates#/media/File:David_-_The_Death_of_Socrates.jpg",
			"public_domain": "yes"
		},
		"filename": "wikimedia_death_of_socrates.jpg",
		"regions": [
			{
				"region_attributes": {
					"image_quality": {
						"good_illumination": true
					},
					"name": "Plato",
					"type": "human"
				},
				"shape_attributes": {
					"height": 227,
					"name": "rect",
					"width": 108,
					"x": 174,
					"y": 139
				}
			},
			{
				"region_attributes": {
					"image_quality": {
						"frontal": true,
						"good_illumination": true
					},
					"name": "Socrates",
					"type": "human"
				},
				"shape_attributes": {
					"height": 209,
					"name": "rect",
					"width": 91,
					"x": 347,
					"y": 114
				}
			},
			{
				"region_attributes": {
					"name": "Hemlock",
					"type": "cup"
				},
				"shape_attributes": {
					"cx": 316,
					"cy": 180,
					"name": "ellipse",
					"rx": 17,
					"ry": 12
				}
			}
		],
		"size": -1
	}
}
```

gaieges replied:
I think you're right, but keep mind this is sort of our own format at this point.  Additionally, we have a ton of stuff that already uses this so my suggestion would- lets roll with this for now and use for validation, etc . but in the near future, come up with a more well designed/clear to understand format that can be used for a similar thing.

Thoughts @e271828- ?

e271828- replied:
I'd stick with what we have for now, but we do need to version this format in order to be able to change it cleanly. That appears to be missing.

And again, please replace any references to VGG in this PR, since there is no "VGG format."

todicus replied:
The only time I see consumption of the nested `class_attributes` is 

https://github.com/IntuitionMachines/omnilytics/blob/93d41437266319ce7ed40e4f205afa99a290353f/match_jobs/job_deployer.py#L92

gaieges replied:
Yep, will change the name and add version

It's also produced by the interchange handler (right?) and consumed by the results page that we're working on in dashboard.

todicus replied:
> It's also produced by the interchange handler (right?) and consumed by the results page that we're working on in dashboard.

Not the nested `class_attributes`. As far as I can tell that showed up when we started trying to parse ImageNet

gaieges replied:
Poking through some ILAS and ILB results, that looks to be right.  I'll toss that (it's flexible anyway as its just checking for string keys), and add a version.  

The intent here was to just add this as a starting point anyway, theres a ways to go still

gaieges replied:
Resolved these comments for when you guys have a second to take another pass

e271828- commented:
This looks mostly fine now, only thing I'd suggest is version be an int or float, not a string.

gaieges commented:
Updated to use integer, also added a test for the no-inner-`class_attributes` approach so that we don't break that down the road; but this model will support both

todicus commented on test.py:
```diff
@@ -341,6 +341,69 @@ def test_webhook(self):
+                "metadata": {
+                    "filename": "image.jpg"
+                },
+                "class_attributes": {
```
Definitely looks cleaner. I still think these labels apply to the whole image, and not an object class within. I'm not clear on the origin of the `class_attributes` field name, actually. Anyway, out of scope of this PR, for now I'm just happy the `class_attributes` aren't procreating!

gaieges replied:
That's how I'm holding it too, I think this single flat `class_attributes` field within the datapoint should address the labels applying to the entire image; is that now how you're consuming this key?

todicus replied:
Yes, that is the functional purpose, at least for ILB. I just think the field name is misleading.

gaieges replied:
How come?  Should it be `file_attributes`?

todicus replied:
`file_attributes` matches VIA format.

gaieges replied:
I see what you mean. Yeah agreed; we can add that into this format no problem, maybe as a v2.
